modules/tensor_mechanics/include/kernels/ADInertialForceShell.h

```diff
@@ -0,0 +1,186 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "ADTimeKernel.h"
+#include "Material.h"
+#include "RankTwoTensor.h"
+#include "libmesh/dense_vector.h"
+
+// Forward Declarations
+class ADInertialForceShell;
+
+namespace libMesh
+{
+class QGauss;
+}
+
+class ADInertialForceShell : public ADTimeKernel
+{
+public:
+  static InputParameters validParams();
+
+  ADInertialForceShell(const InputParameters & parameters);
+
+protected:
+  virtual ADReal computeQpResidual() override { return 0.0; };
+
+  virtual void computeResidual() override;
+  virtual void computeJacobian() override;
+  virtual void computeADOffDiagJacobian() override;
+  virtual void computeShellInertialForces(const MooseArray<ADReal> & _ad_coord,
+                                          const MooseArray<ADReal> & _ad_JxW);
+  virtual void computeResidualForJacobian(ADDenseVector & residual);
+
+private:
+  /// Number of coupled rotational variables
+  unsigned int _nrot;
+
+  /// Number of coupled displacement variables
+  unsigned int _ndisp;
+
+  /// Variable numbers corresponding to rotational variables
+  std::vector<unsigned int> _rot_num;
+
+  /// Variable numbers corresponding to displacement variables
+  std::vector<unsigned int> _disp_num;
+
+  /// Variable numbers corresponding to velocity aux variables
+  std::vector<unsigned int> _vel_num;
+
+  /// Variable numbers corresponding to acceleraion aux variables
+  std::vector<unsigned int> _accel_num;
+
+  /// Variable numbers corresponding to rotational velocity aux variables
+  std::vector<unsigned int> _rot_vel_num;
+
+  /// Variable numbers corresponding to rotational acceleration aux variables
+  std::vector<unsigned int> _rot_accel_num;
+
+  /**
+   * Rotational transformation from global to initial beam local
+   * coordinate system
+   **/
+  ADRankTwoTensor _original_local_config;
+
+  /// Direction along which residual is calculated
+  const unsigned int _component;
+
+  /**
+   * Current translational and rotational velocities at the two nodes
+   * of the beam in the global coordinate system
+   **/
+  ADRealVectorValue _vel_0, _vel_1, _rot_vel_0, _rot_vel_1;
+  ADRealVectorValue _vel_2, _vel_3, _rot_vel_2, _rot_vel_3;
```

Comment:
The comment will only be applied to the first declatation of member variables. I suggest to put all in one line, or all in separate lines and then use the

```
/// @{ Comment that applies to all members below, until the closing comment
Type1 member1;
Type2 member2;
...
///@}
```